src/main/java/org/sonarlint/intellij/issue/persistence/StringStoreIndex.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.issue.persistence;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import org.sonarlint.intellij.proto.Sonarlint;
+
+class StringStoreIndex implements StoreIndex<String> {
+  public static final String INDEX_FILENAME = "index.json";
+  private final Path storeBasePath;
+  private final Path indexFilePath;
+
+  public StringStoreIndex(Path storeBasePath) {
+    this.storeBasePath = storeBasePath;
+    this.indexFilePath = storeBasePath.resolve(INDEX_FILENAME);
+  }
+
+  @Override
+  public Collection<String> keys() {
+    return load().keySet();
+  }
+
+  private Map<String, String> load() {
+    if (!Files.exists(indexFilePath)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this with a call to the "toFile().exists" method [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS3725)